enso-org/enso
# Refactor `LocalStorage` and associated hooks

### Pull Request Description
- Close https://github.com/enso-org/cloud-v2/issues/1586
  - Refactor `localStorage` to use functionality returned by a single function, rather than registering keys plugin-style when modules are loaded.
    - This avoids errors when a module tries to read a key when the module defining the key has not yet been imported.

### Important Notes
None

### Testing Instructions
- Make sure local storage keys are loaded and saved as appropriate:
  - ToS and Privacy Policy modal should not reappear on reload
  - Displayed columns
  - Sidebar open state
  - Local root directory (accessible from gear icon beside "Local" category)
  - Extra local root directories (below the "Local" category in the sidebar - added by "add folder")

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
      or the Snowflake database integration, a run of the [Extra Tests](https://github.com/enso-org/enso/actions/workflows/extra-nightly-tests.yml) has been scheduled.~~
  - ~~If applicable, it is suggested to paste a link to a successful run of the Extra Tests.~~
